src/Module/Create.php

```diff
@@ -134,8 +134,9 @@ private function createConfigProvider(string $modulePath, string $moduleName) :
 
     private function createTemplateNamespace(string $moduleName) : string
     {
-        $namespace = str_replace('\\', '-', $moduleName);
-        $namespace = strtolower($namespace);
-        return $namespace;
+        $pattern     = ['#(?<=(?:\p{Lu}))(\p{Lu}\p{Ll})#', '#(?<=(?:\p{Ll}|\p{Nd}))(\p{Lu})#'];
```

Comment:
I think these regular expressions are not very clear. Maybe will be worth to add some comment here? Why we need two? Is it the simplest way to do what we need? Not sure, but maybe will be better to use `CamelCaseToDash` filter from `zend-filter`?

Reply:
If we comment this an use `zend-filter` instead, we'll need to refactor the `TemplateResolutionTrait.php` also.

Reply:
And if we separate it to another trait, where will that live? I suppose we'll need a `Trait` directory for common traits. I'll wait for @weierophinney to weigh in.